x-pack/platform/plugins/shared/observability_ai_assistant/server/service/knowledge_base_service/reindex_knowledge_base.ts

```diff
@@ -0,0 +1,141 @@
+  logger: Logger;
+  esClient: {
+    asInternalUser: ElasticsearchClient;
+  };
+}): Promise<void> {
+  try {
+    const originalIndex = `${resourceNames.aliases.kb}-000001`;
+    const tempIndex = `${resourceNames.aliases.kb}-000002`;
+
+    const indexSettings = await esClient.asInternalUser.indices.getSettings({
+      index: originalIndex,
+    });
+
+    const createdVersion = parseInt(
+      indexSettings[originalIndex]?.settings?.index?.version?.created ?? '',
+      10
+    );
+
+    // Check if the index was created before version 8.11
+    const versionThreshold = 8110000; // Version 8.11.0
+    if (createdVersion >= versionThreshold) {
+      logger.warn(
+        `Knowledge base index "${originalIndex}" was created in version ${createdVersion}, and does not require reindexing. Semantic text field is already supported.`
+      );
+      return;
+    }
+
+    logger.info(
+      `Knowledge base index was created in ${createdVersion} and must be re-indexed in order to support semantic_text field. Re-indexing now...`
+    );
+
+    // Prevent writes to original index
+    await esClient.asInternalUser.indices.putSettings({
+      index: originalIndex,
+      body: { settings: { 'index.blocks.write': true } },
+    });
+
+    // Create temporary index
+    await esClient.asInternalUser.indices.create({ index: tempIndex });
+
+    // Perform reindex to temporary index
+    await esClient.asInternalUser.reindex({
+      body: {
+        source: { index: originalIndex },
+        dest: { index: tempIndex },
+      },
+      refresh: true,
+      wait_for_completion: true,
+    });
+
+    // Delete original index
+    await esClient.asInternalUser.indices.delete({
+      index: originalIndex,
+    });
+
+    // Prevent writes to temporary index
+    await esClient.asInternalUser.indices.putSettings({
+      index: tempIndex,
+      body: { settings: { 'index.blocks.write': true } },
+    });
+
+    // Re-create original index
+    await esClient.asInternalUser.indices.create({
+      index: originalIndex,
+    });
+
+    // Perform reindex back to original index
+    await esClient.asInternalUser.reindex({
+      body: {
+        source: { index: tempIndex },
+        dest: { index: originalIndex },
+      },
+      refresh: true,
+      wait_for_completion: true,
+    });
+
+    // Delete temporary index
+    await esClient.asInternalUser.indices.delete({
+      index: tempIndex,
+    });
+
+    // Re-create aliases
+    await esClient.asInternalUser.indices.updateAliases({
+      body: {
+        actions: [
+          {
+            add: {
+              index: originalIndex,
+              alias: resourceNames.aliases.kb,
+              is_write_index: true,
+            },
+          },
+        ],
+      },
+    });
+
+    logger.info(
+      'Reindexing knowledge base completed successfully. Semantic text field is now supported.'
+    );
+  } catch (error) {
+    throw new Error(`Failed to reindex knowledge base: ${error.message}`);
```

Comment:
nit: suggesting for consistency

```suggestion
    throw new Error(`Failed to re-index knowledge base: ${error.message}`);
```